R/sim_linelist.R

```diff
@@ -56,9 +56,16 @@
 #' confirmed case and `NA` otherwise for each case in the line list.
 #' Default is `TRUE`. Ct refers to the Cycle threshold from a Real-time
 #' PCR or quantitative PCR (qPCR).
-#' @param min_outbreak_size A single `numeric` defining the minimum chain size
-#' for the simulated outbreak. Default is `10`. This can be increased when the
-#' function should simulate a larger outbreak.
+#' @param outbreak_size A `numeric` vector of length 2 defining the minimum and
+#' the maximum number of infected individuals for the simulated outbreak.
+#' Default is `c(10, 1e5)`, so the minimum outbreak size is 10 infected
+#' individuals, and the maximum outbreak size is 100,000 infected individuals.
+#' Either number can be changed to increase or decrease the maximum or minimum
+#' outbreak size to allow simulating larger or smaller outbreaks. If the
+#' minimum outbreak size cannot be reached after running the simulation for
+#' many iterations (internally) then the function errors, whereas if the
+#' maximum outbreak size is exceeded the function returns the data early and a
+#' warning stating how many cases and contacts are returned.
```

Comment:
I've run it and the warning kind of makes it sound like it's to do with the user rather than the package in itself, maybe a warning saying "In this simulation/iteration, the number of cases exceeded the preferred maximum outbreak size, so data was returned early with X cases and X contacts" so it's more like an explanation than something that's happening because of the numbers provided by the user

Reply:
I think we disagree on this point. If the parameters given to the function by users simulate an outbreak that exceeds the maximum number of cases this might be because of stochasticity providing a simulation with improbably many cases, but it is more likely the case that the user provided parameters that result in the reproduction number being well above 1 and resulting in a huge outbreak. I would say this is the fault of the user for not fully thinking through the parameterisation they are providing, and the current warning is beneficial as it catches these cases for the user while also warning them that the simulation (as currently parameterised) is creating large outbreaks.

This is partially a limit of {simulist} as we currently do not provide any mechanism for finite populations with depletion of susceptible individuals, but it is still for the user to think about how they are running the simulation.

Reply:
As far as I can see, this is the content in the documentation that covers this: "To simulate a line list for COVID-19 with an Poisson contact distribution with a mean number of contacts of 2 and a probability of infection per contact of 0.5, we use the [sim_linelist()](https://epiverse-trace.github.io/simulist/reference/sim_linelist.html) function. The mean number of contacts and probability of infection determine the outbreak reproduction number, if the resulting reproduction number is around one it means we will likely get a reasonably sized outbreak (10 - 1,000 cases, varying due to the stochastic simulation). Take care when setting the mean number of contacts and the probability of infection, as this can lead to the outbreak becoming extremely large."
Maybe I've missed it but I don't see content where the mathematical relationship between the 3 arguments and the resulting R number is explained? I'd suggest doing this, and also adding more examples of how changing these variables results in different reproduction numbers, e.g., what to do to get an R = 1.5, 2, 2.5 etc, so it's understandable 
If a user wants to simulate an epidemic, it's almost guaranteed that they will want a R > 1, so if they try to do so and systematically run into a problem where they're not able to run the simulation I think this will affect the usability of the tool. For instance when I was trying to do this, I only changed the probability of infection from 0.5 to 0.6, and was unable to get a linelist until I was able to set a maximum size for the outbreak. This change doesn't seem big enough from the default for this to happen, especially if I'm not aware of which reproduction number was used in the resulting simulation.
This is part of the reason why using the reproduction number directly would be more intuitive, unless the documentation is more explicit about how it's being obtained.

Reply:
I've expanded the descriptions of the contact distribution and probability of infection and how they result in the reproduction number in the Get Started vignette (28e25b67b6e285034333144314800b8df0c83fe5) and the README (aa812a85191b061e95ee88a23bcbeeb9e6e0e174). 

I will resolve this conversation and merge this PR once the CI checks have passed. If you feel that the descriptions are still not communicating information to the user that is required please make an issue and I can address in a separate PR.